hoodie-common/src/main/java/com/uber/hoodie/common/table/log/block/HoodieAvroDataBlock.java

```diff
@@ -47,10 +51,16 @@
 
   private List<IndexedRecord> records;
   private Schema schema;
+  private Map<HoodieMetadataType, String> metadata;
```

Comment:
should metadata live inside `HoodieLogBlock` with it turned into an abstract class or as default variables? if all blocks are going to have it? seems like a nicer way to do this. that way you can just call `super.getBytes()` etc back and forth?

Reply:
Agreed, more clean that way, but can't get `super.metadataBytes()` to work without some other re-factoring since multiple static code blocks invoke reading of bytes.